test/getHashDigest.test.js

```diff
@@ -6,23 +6,23 @@ describe("getHashDigest()", () => {
   [
     ["test string", "xxhash64", "hex", undefined, "e9e2c351e3c6b198"],
     ["test string", "xxhash64", "base64", undefined, "6eLDUePGsZg="],
-    ["test string", "xxhash64", "base52", undefined, "byfYGDmnmyUr"],
+    ["test string", "xxhash64", "base52", undefined, "rUymnmDGYfyb"],
     ["abc\\0♥", "xxhash64", "hex", undefined, "4b9a34297dc03d20"],
     ["abc\\0💩", "xxhash64", "hex", undefined, "86733ec125b93904"],
     ["abc\\0💩", "xxhash64", "base64", undefined, "hnM+wSW5OQQ="],
     ["abc\\0♥", "xxhash64", "base64", undefined, "S5o0KX3APSA="],
-    ["abc\\0💩", "xxhash64", "base52", undefined, "cfByjQcJZIU"],
-    ["abc\\0♥", "xxhash64", "base52", undefined, "qdLyAQjLlod"],
+    ["abc\\0💩", "xxhash64", "base52", undefined, "UIZJcQjyBfca"],
+    ["abc\\0♥", "xxhash64", "base52", undefined, "dolLjQAyLdqa"],
 
     ["test string", "md4", "hex", 4, "2e06"],
     ["test string", "md4", "base64", undefined, "Lgbt1PFiMmjFpRcw2KCyrw=="],
-    ["test string", "md4", "base52", undefined, "egWqIKxsDHdZTteemJqXfuo"],
+    ["test string", "md4", "base52", undefined, "oufXqJmeetTZdHDsxKIqWge"],
     ["abc\\0♥", "md4", "hex", undefined, "46b9627fecf49b80eaf01c01d86ae9fd"],
     ["abc\\0💩", "md4", "hex", undefined, "45aa5b332f8e562aaf0106ad6fc1d78f"],
     ["abc\\0💩", "md4", "base64", undefined, "RapbMy+OViqvAQatb8HXjw=="],
     ["abc\\0♥", "md4", "base64", undefined, "Rrlif+z0m4Dq8BwB2Grp/Q=="],
-    ["abc\\0💩", "md4", "base52", undefined, "dtXZENFEkYHXGxOkJbevPoD"],
-    ["abc\\0♥", "md4", "base52", undefined, "fYFFcfXRGsVweukHKlPayHs"],
+    ["abc\\0💩", "md4", "base52", undefined, "DoPvebJkOxGXHYkEFNEZXtd"],
+    ["abc\\0♥", "md4", "base52", undefined, "sHyaPlKHkuewVsGRXfcFFYf"],
```

Comment:
So we have broken algorithm?

Reply:
Yup. Unfortunately, changing the hashing to fix the distribution also changes the resulting hashes (for anything except `base64` and `hex`).

<details>
<summary>Same uniformity tests ran on current master</summary>
<img src="https://user-images.githubusercontent.com/810582/140978027-c94d6d24-4527-40af-b33e-77aa2e0bf78f.png">
</details>

Maybe we'd better abandon this PR for the good? Surely enough the distribution is better now, but it may be considered a too-much for a maintenance release.

Reply:
Don't worry about changing hashes, I just want to clarify what our previously implementation was wrong, in near future I fixed the same problems

Reply:
Here's in a nutshell:

If you pick a byte value at (pseudo) random and write it down as a decimal you'll most probably have a leading digit `1` as there are 111 numbers out of possible 256 that are starting with `1`: `1xx`, `1x` and `1`. And you'll almost never get a zero.

But the last digit of the same number is much harder to guess.

Same happened here except the random value is in much bigger range and the base is not 10.

This fix is essentially reversing the order so the resulting substring contains more uniformly distributed "digits" after being trimmed (with some extra care for zero and not extracting the chars that would be cropped anyway)

Reply:
Thanks for clarify

Reply:
On the second thought, there's no need in actually reversing the order. Just slicing the string from it's tail would work as well. 😅

I'll update this PR tomorrow with less dramatic hash updates

Reply:
Feel free to ping me